# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/setuptools/fix-platlibdir.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Fixes setutools to use sys.platlibdir instead of hardcoding /lib/ to
install packages.

  - NoTag <notag@t2sde.org>

--- setuptools-68.0.0/setuptools/command/easy_install.py.vanilla	2023-09-05 20:56:15.665650131 +0200
+++ setuptools-68.0.0/setuptools/command/easy_install.py	2023-09-05 21:00:04.778656017 +0200
@@ -230,7 +230,7 @@
             'sys_exec_prefix': self.config_vars['exec_prefix'],
             # Only python 3.2+ has abiflags
             'abiflags': getattr(sys, 'abiflags', ''),
-            'platlibdir': getattr(sys, 'platlibdir', 'lib'),
+            'platlibdir': sys.platlibdir,
         })
         with contextlib.suppress(AttributeError):
             # only for distutils outside stdlib
@@ -1332,7 +1332,7 @@
 
     INSTALL_SCHEMES = dict(
         posix=dict(
-            install_dir='$base/lib/python$py_version_short/site-packages',
+            install_dir='$base/' + sys.platlibdir + '/python$py_version_short/site-packages',
             script_dir='$base/bin',
         ),
     )
@@ -1531,11 +1531,11 @@
     """Get exe->egg path translations for a given .exe file"""
 
     prefixes = [
-        ('PURELIB/', ''),
+        ('PURELIB/', sys.platlibdir),
         ('PLATLIB/pywin32_system32', ''),
-        ('PLATLIB/', ''),
+        ('PLATLIB/', sys.platlibdir),
         ('SCRIPTS/', 'EGG-INFO/scripts/'),
-        ('DATA/lib/site-packages', ''),
+        ('DATA/' + sys.platlibdir + '/site-packages', ''),
     ]
     z = zipfile.ZipFile(exe_filename)
     try:
--- setuptools-68.0.0/setuptools/_distutils/command/install.py.vanilla	2023-09-05 19:47:47.557544589 +0200
+++ setuptools-68.0.0/setuptools/_distutils/command/install.py	2023-09-05 19:51:41.803550607 +0200
@@ -34,7 +34,7 @@
 
 INSTALL_SCHEMES = {
     'posix_prefix': {
-        'purelib': '{base}/lib/{implementation_lower}{py_version_short}/site-packages',
+        'purelib': '{base}/{platlibdir}/{implementation_lower}{py_version_short}/site-packages',
         'platlib': '{platbase}/{platlibdir}/{implementation_lower}'
         '{py_version_short}/site-packages',
         'headers': '{base}/include/{implementation_lower}'
@@ -43,7 +43,7 @@
         'data': '{base}',
     },
     'posix_home': {
-        'purelib': '{base}/lib/{implementation_lower}',
+        'purelib': '{base}/{platlibdir}/{implementation_lower}',
         'platlib': '{base}/{platlibdir}/{implementation_lower}',
         'headers': '{base}/include/{implementation_lower}/{dist_name}',
         'scripts': '{base}/bin',
